src/firefly/js/visualize/ImageViewerLayout.jsx

```diff
@@ -270,59 +259,58 @@ ImageViewerLayout.propTypes= {
 
 function isImageOnScreen(plotView) {
 
-    const {left,top,scrollViewWidth, scrollViewHeight, viewDim}=  getPositionInfo(plotView);
-    const visible= (left+scrollViewWidth>=0 && left<=viewDim.width && top+scrollViewHeight>=0 && top<=viewDim.height);
-    return visible;
-}
-
+    const {viewDim}= plotView;
+    const plot= primePlot(plotView);
 
-function getPositionInfo(plotView) {
-    var plot= primePlot(plotView);
-    var {dim:{width:viewPortWidth,height:viewPortHeight},x:vpX,y:vpY}= plot.viewPort;
-    var {width:sw,height:sh}= plot.screenSize;
-    var scrollViewWidth= Math.min(viewPortWidth,sw);
-    var scrollViewHeight= Math.min(viewPortHeight,sh);
-    var {scrollX,scrollY, viewDim}= plotView;
-    var left= vpX-scrollX;
-    var top= vpY-scrollY;
-
-    return {left,top,scrollViewWidth, scrollViewHeight, scrollX,scrollY, viewDim};
+    const cc= CysConverter.make(plot);
+    const {screenSize}= plot;
+    const devAsScreenAry= [
+        cc.getScreenCoords(makeDevicePt(0,0)),
+        cc.getScreenCoords(makeDevicePt(viewDim.width,0)),
+        cc.getScreenCoords(makeDevicePt(viewDim.width,viewDim.height)),
+        cc.getScreenCoords(makeDevicePt(0,viewDim.height)),
+        cc.getScreenCoords(makeDevicePt(viewDim.width/2,viewDim.height/2)),
+    ];
+
+
+    const screenAsDevAry= [
+        cc.getDeviceCoords(makeScreenPt(0,0)),
+        cc.getDeviceCoords(makeScreenPt(screenSize.width,0)),
+        cc.getDeviceCoords(makeScreenPt(screenSize.width,screenSize.height)),
+        cc.getDeviceCoords(makeScreenPt(0,screenSize.height)),
+        cc.getDeviceCoords(makeScreenPt(screenSize.width/2,screenSize.height/2)),
+    ];
+
+    let found=  devAsScreenAry.some( (pt) =>
+                   contains(0,0,screenSize.width, screenSize.height,pt.x,pt.y)  ||
+                   intersects(0,0,screenSize.width, screenSize.height,pt.x,pt.y,1,1));
+
+    if (!found) {
+        found= screenAsDevAry.some( (pt) => contains(0,0,viewDim.width, viewDim.height,pt.x,pt.y) ||
+                                            intersects(0,0,viewDim.width, viewDim.height,pt.x,pt.y,1,1));
+    }
```

Comment:
I will write a description.  I had not because it took me awhile to convince myself the concept worked.